storybookjs/storybook
# CSF3: Add auto-title support to on-demand V7/V6 refactor

Issue: #15714 #16059 #16075

## What I did

Refactored to add auto-title to:
- [x] Story store V7 mode
- [x] Story store v6 compatibility mode

Also:
- [x] Removed spurious warnings
- [x] Deprecated `previewCsfV3` feature flag

## How to test

See attached `react-ts` example


If your answer is yes to any of these, please make sure to include it in your PR.
